main/adapters/repository/sqlite_db.go

```diff
@@ -0,0 +1,267 @@
+package repository
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"github.com/google/uuid"
+	"github.com/mattn/go-sqlite3"
+	log "github.com/sirupsen/logrus"
+	"github.com/ubirch/ubirch-client-go/main/config"
+	"github.com/ubirch/ubirch-client-go/main/ent"
+	"github.com/ubirch/ubirch-client-go/main/vars"
+	"time"
+)
+
+// Copyright (c) 2019-2020 ubirch GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// DatabaseManagerSqlite contains the postgres database connection, and offers methods
+// for interacting with the database.
+type DatabaseManagerSqlite struct {
+	options *sql.TxOptions
+	db      *sql.DB
+}
+
+// Ensure Database implements the ContextManager interface
+var _ ContextManager = (*DatabaseManagerSqlite)(nil)
+
+// NewSqlDatabaseInfo takes a database connection string, returns a new initialized
+// database.
+func NewSqliteDatabaseInfo(conf config.Config) (*DatabaseManagerSqlite, error) {
+	db, err := sql.Open(vars.Sqlite, "file:./foo.db?_journal=WAL&mode=rwc&_txlock=exclusive")
+	if err != nil {
+		return nil, err
+	}
+
+	db.SetMaxOpenConns(100)
+	db.SetMaxIdleConns(70)
+	db.SetConnMaxLifetime(10 * time.Minute)
+	if err = db.Ping(); err != nil {
+		return nil, err
+	}
+
+	log.Print("preparing sqlite usage")
+
```

Comment:
table is not created here.

Reply:
i ve sticked to the postgres db. Since we create the tables via migrate.
If we decide to create the tables from here we would need to implement a logic for that.

Reply:
can you please move the table creation from migration into the init method? the init method is called at migration as well, but we need the table to be created for new users as well

Reply:
done